### PR TITLE
musikcube: fix Darwin build

### PR DESCRIPTION
**Pull request by afh**
###### Description of changes

ℹ️ This PR should be reviewed with the changes from #168169 in mind and merged only after #168169 has been merged.
**Update 20220430:** #168169 has been merged and this PR has been rebased onto master containing the changes from #168169.

* Add platform specific dependencies to `buildInputs` using `lib.options stdenv.is$PLATFORM`
* Add Darwin specific `CMakeLists.txt` patch to ensure `ncurses` from nixpkgs is found

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] **No packages depend on musikcube.** Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aanderse** [comment]
After we merge your other PR we can rebase this one and merge. Thanks!

**Comment by afh** [comment]
Sounds good, @aanderse. Thanks for the review!

**Comment by afh** [comment]
Kindly requesting re-review, @anderse, now that #168169 has been merged and this PR is rebased on top of master containing those changes.

**Comment by aanderse** [comment]
Sorry for the delay. Again, we do not need release notes for this... packages break so often on Darwin our changelog would take too long to read if we listed every fix 😉

**Comment by afh** [comment]
All good :) Thanks for the re-review, @aanderse. I've rebased this PR's changes onto master and removed the release notes as requested.

**Comment by afh** [comment]
Thanks for merging @SuperSandro2000 and thanks once again for reviewing and assisting with this PR, @aanderse, much appreciated :pray:

**Comment by Gaelan** [comment]
This MR seems to have broken musikcube on Linux:

```
-- Checking for module 'libsystemd'
--   No package 'libsystemd' found
CMake Error at /nix/store/483fbrh8qyby7ivqmvz09k54976zia92-cmake-3.22.3/share/cmake-3.22/Modules/FindPkgConfig.cmake:603 (message):
  A required package was not found
Call Stack (most recent call first):
  /nix/store/483fbrh8qyby7ivqmvz09k54976zia92-cmake-3.22.3/share/cmake-3.22/Modules/FindPkgConfig.cmake:825 (_pkg_check_modules_internal)
  src/plugins/mpris/CMakeLists.txt:6 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "/build/source/build/CMakeFiles/CMakeOutput.log".
```

I have no idea why, though - as far as I can tell, you didn't change anything about how systemd gets passed in.

**Comment by afh** [comment]
Thanks for raising this, @Gaelan, and apologies that this wasn't caught earlier. This is my first contribution to nixpkgs, if there is anything I could do better in the future to ensure cross-platform stability and avoid breaking changes on system I do not have access to, please let me know :pray:

Mind including the `/build/source/build/CMakeFiles/CMakeOutput.log` for inspection, that would be helpful.

First things that come to my mind:

* Could #168169 already have broken the build?
* Should the [`./0001-apple-cmake.patch`](https://github.com/NixOS/nixpkgs/pull/168173/files#diff-42e5fe038cc196f02aa39302b8687810c3362c8cb74253b00dc4e9c63e02a4c0) be only applied on Darwin? It shouldn't affect how CMake runs on Linux, but who knows…

File: pkgs/applications/audio/musikcube/default.nix [hunk 1]
```diff
@@ -54,9 +59,14 @@ stdenv.mkDerivation rec {
     libev
     libmicrohttpd
     ncurses
-    pulseaudio
     taglib
-  ] ++ lib.optional systemdSupport systemd;
+  ] ++ lib.optional systemdSupport [
```
Comment (Artturin) [inline pkgs/applications/audio/musikcube/default.nix:63]: These should be lib.optionals

Comment (afh) [reply pkgs/applications/audio/musikcube/default.nix:63]: D'oh! 🤦 Thanks @Artturin! I've put #172990 up for review to address this.
I do see uses of `lib.optional` is several other `default.nix` files, I should revisit the Nix documentation to understand the difference.

**Pull request merged**